src/linearization.jl

```diff
@@ -22,7 +23,45 @@ Sparse matrix `A`.
 
 See references [1] and [2] of CARLIN.md.
 """
-function build_matrix(F₁, F₂, N)
+function build_matrix(F₁, F₂, N; compress=false)
+    F₁ = sparse(F₁)
+    F₂ = sparse(F₂)
+    if compress
+        n = size(F₁)[1]
+        monoms = generate_monomials(n, N)
+        nonzero_monoms = firstrest(monoms)[2]
+        monom_to_ind = Dict(m => i for (i, m) in enumerate(nonzero_monoms))
+        result = spzeros(length(monoms) - 1, length(monoms) - 1)
+        for (ind, m) in enumerate(nonzero_monoms)
+            for (i, j, c) in zip(findnz(F₁)...)
```

Comment:
`findnz(F₁)` seems like a loop invariant to me so I suggest having a new variable for it before the main loop